docs/man/man3/Config.3
.TH "Config" 3 "22 Mar 2009" "Version 0.1" "m4php5" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Config \- 
.SH SYNOPSIS
.br
.PP
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static \fBload\fP ($array)"
.br
.ti -1c
.RI "static \fBgetAlternateLangs\fP ()"
.br
.ti -1c
.RI "static \fBgetAllLangs\fP ()"
.br
.ti -1c
.RI "static \fBget\fP ($var)"
.br
.ti -1c
.RI "static \fBset\fP ($var, $val)"
.br
.ti -1c
.RI "static \fBgetPref\fP ($var)"
.br
.ti -1c
.RI "static \fBarrayize\fP (&$item, $key)"
.br
.ti -1c
.RI "static \fBsetPref\fP ($var, $val)"
.br
.in -1c
.SH "Detailed Description"
.PP 
Global configuration helper Store and retrieve configuration options accross application 
.SH "Member Function Documentation"
.PP 
.SS "static Config::load ($ array)\fC [static]\fP"
.PP
Load configuration
.PP
public 
.PP
\fBParameters:\fP
.RS 4
\fIarray\fP $array ?? 
.RE
.PP

.SS "static Config::getAlternateLangs ()\fC [static]\fP"
.PP
Get Alternate Languages
.PP
public 
.SS "static Config::getAllLangs ()\fC [static]\fP"
.PP
Get all language
.PP
public 
.PP
\fBReturns:\fP
.RS 4
?? ?? 
.RE
.PP

.SS "static Config::get ($ var)\fC [static]\fP"
.PP
Get config value
.PP
public 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP $var Variable to get 
.RE
.PP
\fBReturns:\fP
.RS 4
string Value 
.RE
.PP

.SS "static Config::set ($ var, $ val)\fC [static]\fP"
.PP
Set config value
.PP
public 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP $var Variable to set 
.br
\fIstring\fP $val Value 
.RE
.PP

.SS "static Config::getPref ($ var)\fC [static]\fP"
.PP
Get preferences value
.PP
public 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP $var Variable to get 
.RE
.PP
\fBReturns:\fP
.RS 4
string Value 
.RE
.PP

.SS "static Config::arrayize (&$ item, $ key)\fC [static]\fP"
.PP
??
.PP
public 
.PP
\fBParameters:\fP
.RS 4
\fI??\fP $item ?? 
.br
\fI??\fP $key ?? 
.RE
.PP
\fBReturns:\fP
.RS 4
?? Value 
.RE
.PP

.SS "static Config::setPref ($ var, $ val)\fC [static]\fP"
.PP
Set preferences values
.PP
public 
.PP
\fBParameters:\fP
.RS 4
\fIstring\fP $var Variable to set 
.br
\fIstring\fP $val Value 
.RE
.PP
\fBReturns:\fP
.RS 4
string Value 
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for m4php5 from the source code.


Config.php
<?php
/**
 * M PHP Framework
 *
 * @package      M
 * @subpackage   Config
 * @author       Arnaud Sellenet <demental@sat2way.com>
 * @copyright    Copyright (c) 2003-2009 Arnaud Sellenet
 * @license      http://opensource.org/licenses/lgpl-license.php GNU Lesser General Public License
 * @version      0.1
 */

/**
 * Global configuration helper
 * Store and retrieve configuration options accross application
 */

class Config {

	// @todo explain vars

	/**
	 * associative array that stores application preferences 
	 * Application preferences are configurable in the application's database
	 * This array is used as a basic caching system by Config::getPref()
	 * 
	 * @var $preftab array
	 * @access protected
	 */
	protected static $preftab = array();

	/**
	 *
	 * associative array that stores application configuration values
	 * all the configuration values are stored in this array at runtime, 
	 * using Config::load($arr) in M_Setup::setUpEnv()
	 *
	 * @var		array
	 * @access	protected
	 * @
	 */
	protected static $cfgArr = array();

	/**
	 *
	 * Loads a configuration array
	 * if called several times, consecutive arrays are merged.
	 *
	 * @access	public
	 * @param	array	$array Associative array containing app's configuration values
	 * @static
	 *
	 */
	public static function load($array)
	{

		self::$cfgArr = array_merge(self::$cfgArr,$array);
	}

	/**
	 *
	 * Get Alternate Languages (i.e. all languages that are handled by the application except the default language)
	 *
	 * @access	public
	 * @static
	 *
	 */
	public static function getAlternateLangs() {
		$l=Config::getAllLangs();
		$cur=array_search(DEFAULT_LANG,$l);
		unset($l[$cur]);
		return $l;
	}

	/**
	 *
	 * Get all language
	 *
	 * @access	public
	 * @return	array indexed array of all the languages iso2 codes handled by the application
	 * @static
	 *
	 */
	public static function getAllLangs() {
		return Config::get('installedLangs');
	}

	/**
	 *
	 * Get config value
	 *
	 * @access	public
	 * @param	string	$var	Variable to get
	 * @return	string	Value
	 * @static
	 *
	 */
	public static function get($var) {
		return self::$cfgArr[$var];
	}

	/**
	 *
	 * Set config value
	 *
	 * @access	public
	 * @param	string	$var	Variable to set
	 * @param	string	$val	Value
	 * @static
	 *
	 */
	public static function set($var,$val) {
		self::$cfgArr[$var]=$val;
	}

	/**
	 *
	 * Get preferences value.
	 * Preferences are values stored in the application's database,
	 * in a table called 'preferences' 
	 * allowing administrators to modify them through a web interface
	 *
	 * @access	public
	 * @param	string	$var	Variable to get
	 * @return	string	Value
	 * @static
	 *
	 */
	public static function getPref($var) {
		if(!key_exists($var,self::$preftab)){
			$res = & DB_DataObject::factory('preferences');
			$res->var=$var;
			$res->find(true);
			if($res->type=='array') {
				$temp = explode("\n",$res->val);
				foreach($temp as $k=>$v) {
					$temp2 = explode(':',$v);
					$temp3[trim($temp2[0])] = trim($temp2[1]);
				}
				$res->val = $temp3;
			}
			self::$preftab[$var]=$res->val;
		}
		return self::$preftab[$var];
	}

	/**
	 *
	 * @todo what's this ?? (might be scheduled for deletion)
	 *
	 * @access	public
	 * @param	??		$item	??
	 * @param	??		$key	??
	 * @return	??		Value
	 * @static
	 *
	 */
	public static function arrayize(&$item,$key)
	{
	 # code...
	}

	/**
	 *
	 * Set preferences values - affects database
	 *
	 * @access	public
	 * @param	string	$var	Variable to set
	 * @param	string	$val	Value
	 * @return	string Value or false if the target preference is not found in the SQL table
	 * @static
	 *
	 */
	public static function setPref($var,$val)
	{
		$res = & DB_DataObject::factory('preferences');
		$res->var=$var;
		if(!$res->find(true)) {
			return false;
		}
		$res->val = $val;
		$res->update();
		self::$preftab[$var]=$val;
		return self::$preftab[$var];
	}
}